### PR TITLE
feat: add csscomb utility

### PR DESCRIPTION
**Pull request by jason-capsule42**
# Alaska Airlines Pull Request

**Fixes:** #83

## Summary:

This change adds support for using cssComb to do format enforcement on CSS and SCSS files.

The `.csscomb.json` file included in the change defines the rules for formatting and grouping of CSS properties and values.

It is important to note that this tool does not do any lint enforcement. This tool is purely in place to enforce readability standards to improve development and maintainability.

You can read more about the cssComb tool here: https://github.com/csscomb/csscomb.js/

## Type of change:

Please delete options that are not relevant.

- [x] New capability
- [ ] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate)

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

### CODE REVIEW

**Comment by jason-capsule42** [comment]
@blackfalcon I'm thinking about adding a little snippet to the readme.md file for this. 

I have a separate ticket to add documentation for this to the AuroDocSite but a brief overview in the readme feels appropriate to me. I thought you may have a strong opinion about that so I wanted to get your thoughts first.

File: template/.csscomb.json
Comment (blackfalcon) [inline template/.csscomb.json:167]: These prefix defs probably not needed. There should never be any vendor prefix other than in extreme circumstances.

File: template/README.md
Comment (blackfalcon) [inline template/README.md:100]:
```suggestion
Auro now supports [CSScomb](https://www.npmjs.com/package/csscomb) for style formatting enforcement in addition to the [linting requirements](https://auro.alaskaair.com/webcorestylesheets/linter).
```

File: template/README.md
Comment (blackfalcon) [inline template/README.md:104]: I am having concerns about this as a utility. The Github repo seems to have been abandoned and the repo for the VSCode plugin is archived. 

I have concerns about adopting this and then having to remove it in a year.

**Review by blackfalcon — CHANGES_REQUESTED** [review]

**Comment by blackfalcon** [comment]
@geoffrich I am curious as to your opinion on this. The PR itself is ok, but I am having some concerns about adding support for this utility as the tools don't seem to have much support. The CSSComb site is not responding. There haven't been any updates in 2 years and there are a series of ignored PRs. 

In the doc for this @jason-capsule42 stated that there are plug-ins for IDE and the repo for the VSCode plugin is archived. That is concerning.

File: template/.csscomb.json
Comment (jason-capsule42) [reply template/.csscomb.json:167]: The prefixes inclusion in the sort-order settings has no impact on if someone does or does not write them in their stylesheets. This only makes sure that "if they do" they are sorted correctly. If you don't want someone using the prefixes, then that is better handled by a code review or possibly in a lint rule if that is supported.

File: template/README.md
Comment (jason-capsule42) [reply template/README.md:100]: This wording informs the user that this just happened. Which is correct today but won't be in a few months. I chose wording that reads correctly no matter how far in the future someone looks at the document as we certainly won't be editing this language down the line.

File: template/README.md
Comment (jason-capsule42) [reply template/README.md:104]: The repo is not abandoned - https://github.com/csscomb/csscomb.js - and there are 26,000 weekly installs of the npm CSScomb package.

Comment (blackfalcon) [reply template/README.md:104]: The thing that concerns me is that the last updates to the repo were 2 years ago, and that was to bump to 4.3.

I am not disregarding its popularity. I am concerned about its support.

Comment (jason-capsule42) [reply template/README.md:104]: _**TL;DR** - I don't think we necessarily should be enforcing this with the commits - but I do think it's beneficial to contributors and maintainers to provide the tooling in place to to ease their efforts in achieving our stated requirements and ideal formatting of code._

I don't have any particular attachment to this being an enforced part of the build which is why I didn't advocate for this change to begin with. We don't particularly have to enforce that like we want to with our lint rules. It doesn't ultimately matter during execution if `0px` or `0` was used for a property and it doesn't matter if if the top property is assigned before the position property for a given selector - the browser will still figure it out. I love to have it consistent and I think there is "better (IMO)" way to format the code but I also don't think it's worth opening up to a possible debate with every individual about what the right format to use is or how to go about getting there as this is often an arduous debate. Similarly, I got bored with the tabs vs. spaces and hard G vs soft G in "GIF" almost as soon as those heated debates started. 

We should at a minimum have a style guide to promote the preferred formatting and leave it up to the contributor to decide for themselves how they achieve that goal. If someone wants to do it manually, good for them, that's great. If someone wants to use a utility to simplify the process, great. I happen to use CSScomb to achieve the desired result because it's simple to use and a big time saver. If CSScomb were to stop working at some point I would certainly be disappointed. But it wouldn't change the reason I used it and I would still be striving to format my styles in the same manner one way or another.

By establishing a style guide for Auro - providing a config file for formatting tools that a developer may use is at least a polite courtesy. To me this would mean adding the NPM package, the config file and a script for use with `npm run ...` but not enforce the use of it.

**Comment by jason-capsule42** [comment]
> @geoffrich I am curious as to your opinion on this. The PR itself is ok, but I am having some concerns about adding support for this utility as the tools don't seem to have much support. The CSSComb site is not responding. There haven't been any updates in 2 years and there are a series of ignored PRs.
> 
> In the doc for this @jason-capsule42 stated that there are plug-ins for IDE and the repo for the VSCode plugin is archived. That is concerning.

I actually just discovered that PostCSS has plugins for what we are looking for here:

```
**Better CSS Readability**
- Precss contains plugins for Sass-like features, like variables, nesting, and mixins.
- postcss-sorting sorts the content of rules and at-rules.
- postcss-utilities includes the most commonly used shortcuts and helpers.
- short adds and extends numerous shorthand properties.
```

https://github.com/postcss/postcss

With that in mind I suggest looking at including those PostCSS plugins in place of CSScomb since we already have PostCSS integrated.

**Comment by geoffrich** [comment]
It is concerning that the CSScomb repository appears to be in maintenance mode and their site is down.

+1 for using a PostCSS plugin if possible, since that's already integrated. It seems like a PostCSS plugin would be easier for us to fork and maintain if necessary.

**Pull request closed without merging**
